Fix install_name
https://github.com/redis/hiredis/issues/437
https://github.com/redis/hiredis/pull/595/commits/881fcb776d4237168f9ad4b2dd32c6ea902628ac
Also use -dynamiclib instead of -shared for compatibility with older macOS.
--- Makefile.orig	2015-09-16 05:25:02.000000000 -0500
+++ Makefile	2018-05-14 14:15:38.000000000 -0500
@@ -63,7 +63,7 @@
 ifeq ($(uname_S),Darwin)
   DYLIBSUFFIX=dylib
   DYLIB_MINOR_NAME=$(LIBNAME).$(HIREDIS_SONAME).$(DYLIBSUFFIX)
-  DYLIB_MAKE_CMD=$(CC) -shared -Wl,-install_name,$(DYLIB_MINOR_NAME) -o $(DYLIBNAME) $(LDFLAGS)
+  DYLIB_MAKE_CMD=$(CC) -dynamiclib -Wl,-install_name,$(PREFIX)/$(LIBRARY_PATH)/$(DYLIB_MINOR_NAME) -o $(DYLIBNAME) $(LDFLAGS)
 endif
 
 all: $(DYLIBNAME) $(STLIBNAME) hiredis-test $(PKGCONFNAME)
